provisioning/miniprov/hare_mp/store.py

```diff
@@ -41,3 +42,26 @@ def __init__(self, url: str):
 
     def _raw_get(self, key: str) -> str:
         return self.conf.get('hare', key)
+
+    def get_machine_id(self) -> str:
+        with open('/etc/machine_id', 'r') as f:
+            machine_id = f.readline().strip('\n')
+            hostname = self.conf.get('hare',
+                                     f'server_node>{machine_id}>hostname')
+            if hostname != socket.gethostname():
```

Comment:
It reminds me [why URL.equals() is broken](https://news.ycombinator.com/item?id=21765788) in Java (in short: because using DNS lookups to compare two URLs is not a good idea).

1. Why do we need to compare the hostname via socket.gethostname()
2. Are you sure it is a correct place for this check?

I agree that this functionality has some benefit but I'm a bit concerned about the fact that this method has two separate responsibilities. In other words, I just want to get the machine ID but instead the method will also resolve  the hostname and raise an exception if the hostname is not that good. I didn't ask for hostname, I wanted machine_id only 😁

Reply:
Yes, here we are verifying the machine id we got from conf-store if it is of the same node that we expect. We confirm this by comparing the hostname from conf-store for the given machine-id with the node's hostname. Its just another error case that we are trying to catch that may avoid additional debugging in case of an inconsistent conf-store.

Reply:
I see but can we do that check in a separate method?

Reply:
Okay, separate method makes sense.

Reply:
Done.